sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -815,6 +815,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val HIVE_METASTORE_PARTITION_LIMIT =
+    buildConf("spark.sql.hive.metastorePartitionLimit")
+      .doc("The maximum number of metastore partitions allowed. Use -1 for unlimited.")
```

Comment:
The current statement looks a bit ambiguous, so how about following the statement of the Hive config you pointed out?